lib/new_relic/agent/transaction/abstract_segment.rb

```diff
@@ -327,6 +331,25 @@ def transaction_state
             Tracer.state
           end
         end
+
+        def invoke_callback
+          return unless self.class.instance_variable_defined?(CALLBACK)
+
+          NewRelic::Agent.logger.debug("Invoking callback for #{self.class.name}...")
+          self.class.instance_variable_get(CALLBACK).call
+        end
+
+        def self.set_segment_callback(callback_proc)
+          unless callback_proc.is_a?(Proc)
+            NewRelic::Agent.logger.error("#{self}.#{__method__}: expected an argument of type Proc, " \
+                                         "got #{callback_proc.class}")
+            return
+          end
+
+          label = "set_callback_#{name.split('::').last.downcase.sub(SEGMENT, NewRelic::EMPTY_STR)}".to_sym
+          NewRelic::Agent.record_api_supportability_metric(label)
```

Comment:
I like that we're recording a supportability metric on this. Would be cool to see usage.